src/Attribute/Part.php

```diff
@@ -0,0 +1,32 @@
+<?php
+declare(strict_types=1);
+
+namespace Retrofit\Attribute;
+
+use Attribute;
+use Retrofit\HttpMethod;
+use Retrofit\MimeEncoding;
+
+#[Attribute(Attribute::TARGET_PARAMETER)]
+readonly class Part
+{
+    private MimeEncoding $encoding;
+
+    public function __construct(
+        private ?string $name = null,
+        MimeEncoding|string $encoding = MimeEncoding::BINARY
+    )
+    {
+        $this->encoding = is_string($encoding) ? HttpMethod::from($encoding) : $encoding;
```

Comment:
Maybe all of this duplicated lines should also be in some kind of a util? Or maybe `HttpMethod::from` should handle both string and `MimeEncoding` values? (it is even better I think)

Reply:
`HttpMethod::from` method is a built-in PHP factory method to creating enums from string. If we pass something else than string, exception will be thrown.

I suggest create method in the enum `MimeEncoding`:

```php
public static function fromEnumOrString(MimeEncoding|string $from): MimeEncoding
{
    return is_string($from) ? MimeEncoding::from($from) : $from;
}
```

Then can use it:

```php
$this->encoding = MimeEncoding::fromEnumOrString($encoding);
```

BTW. You found a really nasty bug. We was creating `MimeEncoding` from `HttpMethod`...